### PR TITLE
Leetcode

### PR DESCRIPTION
**Pull request by joshpeng1999**
Leetcode section on the website. Take a look at tablet view and lmk if it's okay

### CODE REVIEW

File: src/index.css [hunk 1]
```diff
@@ -9,6 +10,7 @@
   --font-body-heading: 600 36px Barlow;
```
Comment (praneershrest) [inline src/index.css:10]: This was a mistake on my part. Can you change  `--font-body-heading: 600 36px Barlow` to  `--font-body-heading: 600 30px Barlow`

File: src/App.js [hunk 2]
```diff
@@ -7,9 +8,9 @@ const App = () => {
     <div id={styles.app}>
       <Router>
         <Switch>
-          {/* <Route path="/path">
-            <Component />
-          </Route> */}
+          <Route path="/path">
+            <Leetcode />
```
Comment (praneershrest) [inline src/App.js:12]: Can you change this to the original App.js so we don't get too many merge conflicts :))

File: src/components/LeetcodeText.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import styles from '../css/LeetcodeText.module.css'
```
Comment (praneershrest) [inline src/components/LeetcodeText.js:1]: Add a media query so you can use the useMedia hook.

Import: `import useMediaQuery from "../utils/useMediaQuery`
Defining it: const `isDesktop = useMediaQuery('min-width: 600px')
Check out the other components (Note or UpcomingEvent)

File: src/components/LeetcodeText.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+const LeetcodeText = props => {
+  return (
+    <div className={styles.container}>
+      <p className={styles.title}>Leetcode</p>
```
Comment (praneershrest) [inline src/components/LeetcodeText.js:6]: I think the marginBottom for styles.title is different for desktop and mobile. With the media hook, you can make conditional styling:

`style={isDesktop ? { marginBottom: '7px' } : { marginBottom: '5px' }}`

File: src/components/LeetcodeText.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      <p className={styles.bodyText}>
```
Comment (praneershrest) [inline src/components/LeetcodeText.js:7]: Conditional render: `style={isDesktop ? { lineHeight: '32px' } : { lineHeight: '16px' }}`

File: src/css/LeetcodeText.module.css [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  margin-bottom: 16px;
+  padding-left: 1%;
+  font: var(--font-body-title);
+  color: var(--color-text-white);
+}
+
+.bodyText {
+  margin: 0;
+  margin-bottom: 23px;
+  padding-left: 1%;
```
Comment (praneershrest) [inline src/css/LeetcodeText.module.css:19]: Don't think you need `padding-left: 1%`

File: src/css/LeetcodeText.module.css [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  margin-left: 14%;
+  margin-right: 18%;
+  padding-top: 73px;
+  padding-bottom: 45%;
+}
+
+.title {
+  margin: 0;
+  margin-bottom: 16px;
+  padding-left: 1%;
```
Comment (praneershrest) [inline src/css/LeetcodeText.module.css:11]: Don't think you need padding-left: 1% and can remove margin-bottom if you conditionally render through the js component.

File: src/css/LeetcodeText.module.css [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  font: var(--font-body-title);
```
Comment (praneershrest) [inline src/css/LeetcodeText.module.css:12]: Should be `font: var(--font-body-heading)`

File: src/css/LeetcodeText.module.css [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  height: 43px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background-color: #89aff0;
+  border: 1px solid var(--color-text-white);
+}
+
+.linkContainer {
+  width: 94%;
```
Comment (praneershrest) [inline src/css/LeetcodeText.module.css:39]: Make width: 100% but I think you don't need it if you make it 100%.

**Review by praneershrest — CHANGES_REQUESTED** [review]

File: src/components/Leetcode.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import LeetcodeText from './LeetcodeText.js'
+import laptopMonitor from '../assets/laptop-monitor.svg'
+import styles from '../css/Leetcode.module.css'
+
+const Leetcode = props => {
+  return (
+    <div>
```
Comment (alikhamesy) [inline src/components/Leetcode.js:7]: Unstyled divs can be replaced with `<></>` fragments

**Review by praneershrest — APPROVED** [review]

**Pull request merged**